src/renderer/src/routes/pages/Send/index.tsx

```diff
@@ -1,114 +1,237 @@
-import { useState } from 'react'
+import { Fragment, useCallback, useEffect, useState } from 'react'
 import { useTranslation } from 'react-i18next'
-import { TbArrowDown, TbStepOut } from 'react-icons/tb'
+import { TbArrowDown, TbEye, TbStepOut } from 'react-icons/tb'
+import { Account, BlockchainService, isCalculableFee } from '@cityofzion/blockchain-service'
 import { TokenBalance } from '@renderer/@types/query'
 import { IAccountState } from '@renderer/@types/store'
 import { Button } from '@renderer/components/Button'
 import { Separator } from '@renderer/components/Separator'
-import { useAccountsSelector } from '@renderer/hooks/useAccountSelector'
-import { useBalancesAndExchange } from '@renderer/hooks/useBalancesAndExchange'
+import { DoraHelper } from '@renderer/helpers/DoraHelper'
+import { useBsAggregatorSelector } from '@renderer/hooks/useBlockchainSelector'
+import { useModalNavigate } from '@renderer/hooks/useModalRouter'
+import { useEncryptedPasswordSelector } from '@renderer/hooks/useSettingsSelector'
 import { ContentLayout } from '@renderer/layouts/ContentLayout'
 
 import { Recipient } from './Recipient'
 import { SelectAccount } from './SelectAccount'
 import { SelectToken } from './SelectToken'
 import { SendAmount } from './SendAmount'
-import { TokenBalanceList } from './TokenBalanceList'
 import { TotalFee } from './TotalFee'
 
 enum SendPageStep {
   SelectAccount = 1,
   SelectToken = 2,
   SelectAmount = 3,
   SelectContact = 4,
+  Send = 5,
+}
+
+type TSendServiceResult = {
+  serviceAccount: Account
+  service: BlockchainService
+  token: TokenBalance
 }
 
 export const SendPage = () => {
   const { t } = useTranslation('pages', { keyPrefix: 'send' })
-  const { accounts } = useAccountsSelector()
-  const balanceExchange = useBalancesAndExchange(accounts[0])
   const [selectedAccount, setSelectedAccount] = useState<IAccountState>()
   const [selectedToken, setSelectedToken] = useState<TokenBalance | null>()
   const [selectedAmount, setSelectedAmount] = useState<number>(0)
-  const [currentStep, setCurrentStep] = useState<SendPageStep>(SendPageStep.SelectAccount)
+  const [selectedRecipient, setSelectedRecipient] = useState<string>('')
+  const [currentStep, setCurrentStep] = useState<number>(SendPageStep.SelectAccount)
+  const { modalNavigate } = useModalNavigate()
+  const [totalFee, setTotalFee] = useState<string>('0.00')
+  const { bsAggregatorRef } = useBsAggregatorSelector()
+  const { encryptedPasswordRef } = useEncryptedPasswordSelector()
```

Comment:
One thing I like to do to make the code cleaner and easier to read is to group states, group hooks, group memos. Sometimes this won't be possible because something may need something else, but whenever possible I try to separate them. So I like to leave the hooks first, then the states, and then the memos.

```suggestion
 const { t } = useTranslation('pages', { keyPrefix: 'send' })
 const { bsAggregatorRef } = useBsAggregatorSelector()
 const { encryptedPasswordRef } = useEncryptedPasswordSelector()
  const { modalNavigate } = useModalNavigate()
  
  const [selectedAccount, setSelectedAccount] = useState<IAccountState>()
  const [selectedToken, setSelectedToken] = useState<TokenBalance | null>()
  const [selectedAmount, setSelectedAmount] = useState<number>(0)
  const [selectedRecipient, setSelectedRecipient] = useState<string>('')
  const [currentStep, setCurrentStep] = useState<number>(SendPageStep.SelectAccount)
  const [totalFee, setTotalFee] = useState<string>('0.00')
  
```